### PR TITLE
[Merged by Bors] - doc(Geometry/Manifold/LieGroup): extend and clean up docstrings

### PR DESCRIPTION
**Pull request by grunweg**
- tweak section names; add comments explaining what's in them
- add two lemma docstrings
- revamp the module docstring to mention all results
- in the module docstring, separate main definitions and results

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by j-loreaux** [comment]
@grunweg yes, please remove the autoImplicit commit so that these PRs can just be handled separately, especially since neither on has a `depends on` line in the PR description.

**Comment by grunweg** [comment]
awaiting-review

File: Mathlib/Geometry/Manifold/Algebra/LieGroup.lean [hunk 1]
```diff
@@ -21,10 +21,15 @@ groups here are not necessarily finite dimensional.
 ## Main definitions and statements
 
 * `LieAddGroup I G` : a Lie additive group where `G` is a manifold on the model with corners `I`.
-* `LieGroup I G`     : a Lie multiplicative group where `G` is a manifold on the model with
-                        corners `I`.
+* `LieGroup I G` : a Lie multiplicative group where `G` is a manifold on the model with corners `I`.
 * `normedSpaceLieAddGroup` : a normed vector space over a nontrivially normed field
-                                 is an additive Lie group.
+  is an additive Lie group.
+* point-wise inversion and division of maps `M → G` is smooth
```
Comment (j-loreaux) [inline Mathlib/Geometry/Manifold/Algebra/LieGroup.lean:27]: You should specify which declaration(s) provide this fact

File: Mathlib/Geometry/Manifold/Algebra/LieGroup.lean [hunk 2]
```diff
@@ -21,10 +21,15 @@ groups here are not necessarily finite dimensional.
...
+* `SmoothInv₀`: typeclass for smooth manifolds with `0` and `Inv` such that inversion is a smooth
+  map at each non-zero point. This includes complete normed fields and Lie groups.
+* if `SmoothInv₀ N`, point-wise inversion of smooth maps `f : M → N` is smooth at all points
+at which `f` doesn't vanish. If also `SmoothMul N` (i.e., `N` is a Lie group except possibly
+for smoothness of inversion at `0`), similar results hold for point-wise division.
```
Comment (j-loreaux) [inline Mathlib/Geometry/Manifold/Algebra/LieGroup.lean:32]: also need to specify the declaration here.
```suggestion
* if `SmoothInv₀ N`, point-wise inversion of smooth maps `f : M → N` is smooth at all points
  at which `f` doesn't vanish. If also `SmoothMul N` (i.e., `N` is a Lie group except possibly
  for smoothness of inversion at `0`), similar results hold for point-wise division.
```

File: Mathlib/Geometry/Manifold/Algebra/LieGroup.lean [hunk 3]
```diff
@@ -65,7 +70,10 @@ class LieGroup {𝕜 : Type*} [NontriviallyNormedField 𝕜] {H : Type*} [Topolo
   smooth_inv : Smooth I I fun a : G => a⁻¹
 #align lie_group LieGroup
 
-section LieGroup
+/-! Let `f : M → G` be `C^n` or smooth functions into a smooth Lie group, then `f` is point-wise
+  invertible with smooth inverse `f`. If `f` and `g` are two such functions, the quotient
+  `f / g` (i.e., the point-wise product of `f` and the point-wise inverse of `g` is also smooth. -/
```
Comment (j-loreaux) [inline Mathlib/Geometry/Manifold/Algebra/LieGroup.lean:75]: Can you add a section header, like so? I assume `to_additive` does some of this? IF not, remove the references to negation and subtraction.
```suggestion
/-!
### Smoothness of inversion, negation, division and subtraction

Let `f : M → G` be `C^n` or smooth functions into a smooth Lie group, then `f` is point-wise
  invertible with smooth inverse `f`. If `f` and `g` are two such functions, the quotient
  `f / g` (i.e., the point-wise product of `f` and the point-wise inverse of `g` is also smooth. -/
```

File: Mathlib/Geometry/Manifold/Algebra/LieGroup.lean [hunk 4]
```diff
@@ -231,7 +241,10 @@ instance normedSpaceLieAddGroup {𝕜 : Type*} [NontriviallyNormedField 𝕜] {E
   smooth_neg := contDiff_neg.contMDiff
 #align normed_space_lie_add_group normedSpaceLieAddGroup
 
-section HasSmoothInv
+/-! Typeclass for smooth manifolds with 0 and Inv such that inversion is smooth at all non-zero
```
Comment (j-loreaux) [inline Mathlib/Geometry/Manifold/Algebra/LieGroup.lean:244]: Please add a section heading

File: Mathlib/Geometry/Manifold/Algebra/LieGroup.lean [hunk 5]
```diff
@@ -301,8 +314,10 @@ theorem SmoothOn.inv₀ (hf : SmoothOn I' I f s) (h0 : ∀ x ∈ s, f x ≠ 0) :
     SmoothOn I' I (fun x => (f x)⁻¹) s :=
   ContMDiffOn.inv₀ hf h0
 
-end HasSmoothInv
+end SmoothInv₀
 
+/-! Point-wise division of smooth functions `f : M → G` where `[SmoothMul I G]` and
```
Comment (j-loreaux) [inline Mathlib/Geometry/Manifold/Algebra/LieGroup.lean:319]: one more section heading, please.

**Review by j-loreaux — COMMENTED** [review]
minor comments

**Comment by j-loreaux** [comment]
Please also update the PR description (and title if relevant)

File: Mathlib/Geometry/Manifold/Algebra/LieGroup.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Geometry/Manifold/Algebra/LieGroup.lean:27]: Well, there are about half a dozen versions of each ({Smooth,ContMDiff}{,On,At}{div,.inv}), which is why I didn't do this at first. I've just referenced the first two.

File: Mathlib/Geometry/Manifold/Algebra/LieGroup.lean (hunk 3 above)
Comment (grunweg) [reply Mathlib/Geometry/Manifold/Algebra/LieGroup.lean:75]: Indeed, to_additive generates corresponding lemmas for negation and subtraction.

**Comment by grunweg** [comment]
Thanks for the fast review! I just incorporated all comments and also pushed two more commits tweaking the docs further.
awaiting-review

**Comment by j-loreaux** [comment]
Thanks!

bors merge

**Pull request closed without merging**